### PR TITLE
[SPARK-24111][SQL] Add the TPCDS v2.7 (latest) queries in TPCDSQueryBenchmark

### PR DESCRIPTION
**Pull request by maropu**
## What changes were proposed in this pull request?
This pr added  the TPCDS v2.7 (latest) queries in `TPCDSQueryBenchmark`.
These query files have been added in `SPARK-23167`.

## How was this patch tested?
Manually checked.

### CODE REVIEW

**Comment by maropu** [comment]
@gatorsmile

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala [hunk 1]
```diff
@@ -87,10 +90,20 @@ object TPCDSQueryBenchmark extends Logging {
     }
   }
 
+  def filterQueries(
+      origQueries: Seq[String],
+      args: TPCDSQueryBenchmarkArguments): Seq[String] = {
+    if (args.queryFilter.nonEmpty) {
+      origQueries.filter { case queryName => args.queryFilter.contains(queryName) }
```
Comment (xuanyuanking) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala]:
```
origQueries.filter(args.queryFilter.contains)
```
maybe better? :)

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala [hunk 2]
```diff
@@ -78,7 +81,7 @@ object TPCDSQueryBenchmark extends Logging {
       }
       val numRows = queryRelations.map(tableSizes.getOrElse(_, 0L)).sum
       val benchmark = new Benchmark(s"TPCDS Snappy", numRows, 5)
-      benchmark.addCase(name) { i =>
+      benchmark.addCase(s"$name$nameSuffix") { _ =>
```
Comment (xuanyuanking) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala]: how about 
```
benchmark.addCase(s"$name$nameSuffix")(_ => spark.sql(queryString).collect())
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala (hunk 1 above)
Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala]: Thanks, fixed.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala (hunk 2 above)
Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala]: yes and no; I feel both is ok.

**Comment by maropu** [comment]
ping @gatorsmile

**Review by gatorsmile — COMMENTED** [review]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merged to master

**Pull request closed without merging**